### DISCUSSION TITLE
Prisma client not updating when deployed on digitalocean app platform.

### DISCUSSION THREAD

**Discussion by pbandjs** [prisma/prisma#6165] [Q&A] [answered]
I have a graphql server that uses the dependency @prisma/client:"^2.17.0"

I have successfully deployed it on digitalocean app platform. Until recently, i have made changes to my database using prisma migrate and redployed the server with no issues. However, i recently pushed some pretty straightforward changes, and find that my client is not updated when deployed. 

For instance, after adding a model named EventRegistration, migrating the database, deploying migrations and regenerating the client, queries such as the one below work when running everything locally, but throw the error "Error: Cannot read property 'findMany' of undefined" after the server is successfully redeployed on DigitalOcean app platform.

```
const { PrismaClient } = require("@prisma/client");
const prisma = new PrismaClient();
await ctx.prisma.eventRegistration.findMany({
    where: { event: { id: args.eventId } },
  });
```

I was previously able to make a deploy changes the same way, and they worked as expected. Has anyone faced this before and know of a solution ?

**Comment by 2color** [comment]
Hey @pbandjs 👋,

Can you share the full error that you get from the logs on DigitalOcean's App Platform?

It sounds like this might be due to the Prisma Client not being correctly generated. 

I would try adding `npx prisma generate` to the [build step](https://www.digitalocean.com/docs/app-platform/concepts/command/) on App Platform.

**Reply by pbandjs** [reply]
Hi @2color,

I think the @prisma/client package generates the client in the postinstall hook. My suspicion is that the App Platform caches packages so the postinstall step never runs. When i upgraded my package version and deployed, it installed the package again and the client seems to have generated correctly. I haven't tried your suggestion but if my suspicion is correct, then it should work. Another possible solution I haven't tried is setting the environment variable NODE_MODULES_CACHE to false.

**Reply by 2color** [reply]
Hey @pbandjs,

It sounds like this is probably a caching-related problem. I would still recommend trying to add a build step with `prisma generate` to see if that solves the problem. 

I also [posted a question on the DigitalOcean Q&A](https://www.digitalocean.com/community/questions/how-do-i-disable-caching-for-node_modules-in-the-app-platform-build) to see if there's any workaround for this.

Regarding `NODE_MODULES_CACHE`, I couldn't find this documented anywhere on DigitalOcean, so I doubt this would work (it seems that this env var is a Dokku thing).

**Reply by abdel-ships-it** [reply]
@2color Digitalocean app platform uses [heroku-buildpack-nodejs](https://github.com/heroku/heroku-buildpack-nodejs#readme) under the hood, which uses `NODE_MODULES_CACHE` under the hood. See https://devcenter.heroku.com/articles/nodejs-support#cache-behavior

**Comment by riley-worthington** [comment, marked as answer]
I fixed by adding `prisma generate` to the build script in my package.json.
```
"scripts": {
    "build": "prisma generate && nest build"
}
```
And in Digital Ocean app platform, the build command is still `npm run build`

**Comment by StevenMcgrew** [comment]
For the DigitalOcean App Platform, I had to set the environment variable `NODE_MODULES_CACHE ` to false. Adding the build script with either `prisma generate` or `npx prisma generate` did not fix it.

**Comment by ilgar-zada** [comment]
I had the same problem and resolved it by re-setting Environment variable (DATABASE_URL), which I had set during the creation of application. You can check it from Settings->Environment Variables of your app component.
![image](https://github.com/prisma/prisma/assets/70514557/3c8c0f50-33d9-4da0-bacc-f393f6e5eafe)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
